### ISSUE TITLE
 allow timestamps to create indexes automatically

### ISSUE THREAD

**Issue by ChangePlaces** [laravel/internals#891]
`$table->timestamps()` is a handy way in migrations to create `created_at` and `updated_at` fields. fields that nearly always require an index. It'd be nice if it were possible to do something like `$table->timestampsWithIndexes(['created_at', 'updated_at'])` or `$table->timestampsWithIndexes(['created_at'])` to add indexes

**Comment by drbyte** [comment]
I get the appeal, but I'm not sure that it's any faster.

Today you would use:
```php
$table->timestamps();
$table->index(['created_at', 'updated_at']); // being intentional about which columns to index
```

Your proposal, while slightly more expressive, is only slightly shorter, but still requires naming which columns to index:
```php
$table->timestampsWithIndexes(['created_at', 'updated_at']); // also naming which columns to index
```

While I'm not a fan of creating unnecessary indexes, especially on large tables, I would think if a `timestampsWithIndexes()` method were added, it should default to creating indexes on both columns, and allow for an override such as the array you used, where you name specifically which of the timestamp columns are to be given indexes ... ie: in case you only want 1 instead of both. 

And then there's the softDelete timestamp, not discussed here, but affected by any change in patterns related to the methods used for these schema alterations.


Personally I'm inclined to just specify which indexes I want to create, using `$table->index([column-list]);`, whether for timestamp fields or otherwise. I probably wouldn't use the `timestampsWithIndexes()` approach much.  But that's just my take.

**Comment by sisve** [comment]
> [...] fields that nearly always require an index [...]

How do I tag this with a [citation needed] tag? There's no source for this assumption.

**Comment by reinink** [comment]
I agree with @drbyte, I don't think it makes sense to build in indexes like this by default. Indexes are not free, and they should really only be added when needed for specific queries. And since it's so trivial to add these indexes using the `$query->index()` method, I don't really see the need to add a `$query->timestampsWithIndexes()` method.
